File: medicines-frontend/src/app/models/Stock.ts
import {Medicine} from "./Medicine";
import {Person} from "./Person";

export interface Stock{
    id: string,
    medicine: Medicine,
    validityEndDate: Date,
    validityStatus: ValidityStatusEnum,
    person: Person
}

export enum ValidityStatusEnum{
    IN_VALIDITY = "IN_VALIDITY",
    LESS_THAN_1_MONTH = "LESS_THAN_1_MONTH",
    WITHOUT_VALIDITY = "WITHOUT_VALIDITY"
}

File: medicines/src/main/java/com/nr/medicines/models/Stock.java
package com.nr.medicines.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.UUID;

@Entity
@Table(name = "stock")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Stock {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "uuid")
    private UUID uuid;

    @ManyToOne
    @JoinColumn(name = "medicine_barcode")
    private Medicine medicine;

    @Column(name = "validity_end_date")
    private Date validityEndDate;

    @ManyToOne
    @JoinColumn(name = "person_uuid")
    private Person person;

    @Column(name = "created")
    private Date created;
}
